packages/react/src/components/UIShell/components/styles/_side-nav.scss

```diff
@@ -172,6 +173,75 @@ div:has(.#{$prefix}--header)
     text-align: start;
   }
 
+  &:hover {
+    .#{$prefix}--side-nav__items {
+      overflow: visible;
+    }      
+  }
+
+  .#{$prefix}--popover-container {
+    display: block;
+    width: 100%;
+  }
+
+  .#{$prefix}--side-nav__items {
+    overflow: visible;
+  }  
+
+  .#{$prefix}--side-nav__submenu {
+    position: relative;
+
+    .#{$prefix}--side-nav--panel-submenu-caret-container {
+      position: absolute;
+      width: 100%;
+      height: 100%;
+    }
+  
+    .#{$prefix}--side-nav--panel-submenu-caret {
+      position: absolute;
+      bottom: 4px;
+      right: 4px;
+ 
+      > svg {
+        fill: $icon-secondary;
+      }
+  
+    }  
+  }
+
```

Comment:
Just a general note that we typically avoid nesting selectors in Carbon.
https://github.com/carbon-design-system/carbon/blob/main/docs/developer-handbook.md#coding-style